feat: FIT-652: Introduce Search on Labeling Template Gallery

ricardoantoniocm opened the pull request:
This PR introduces keywords to all labeling config templates.

Keywords include industry applications, models associated with this kind of labeling configuration and general domain terminology.

These keywords power the template search on LSE.

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

ricardoantoniocm committed:
feat: FIT-652: Introduce Search on Labeling Template Gallery

ricardoantoniocm committed:
Decorates templates with keywords.

ricardoantoniocm committed:
Adds keywords to the rest of the templates.

ricardoantoniocm committed:
Adds bbox keyword.

ricardoantoniocm committed:
Refactors keywords to a simple array to avoid modifying backend.

ricardoantoniocm committed:
Refactors all template configs.

ricardoantoniocm committed:
Revert projects/api.py to develop.

ricardoantoniocm committed:
Add gen ai to Generative AI templates.

caitlinwheeless commented on label_studio/annotation_templates/computer-vision/keypoints/config.yml:
```diff
@@ -3,7 +3,16 @@ type: community
 image: /static/templates/keypoints.png
-details: <h1>Click the label and then on the canvas</h1>
+details: |
+  <h1>Click the label and then on the canvas</h1>
```
```suggestion
  <h1>Add labeled keypoints</h1>
```

caitlinwheeless commented on label_studio/annotation_templates/computer-vision/keypoints/config.yml:
```diff
@@ -3,7 +3,16 @@ type: community
+    <dt>Associated Models</dt>
+    <dd>OpenPose, MediaPipe, PoseNet, DensePose, Face++, Dlib</dd>
+    <dt>Domain Terminology</dt>
+    <dd>anatomical landmarks, joint detection, skeletal tracking, facial keypoints, body pose analysis</dd>
```
Maybe we want to reserve our skeletal tracking/pose stuff for vectors, which will be way better at it?

ricardoantoniocm replied:
Hmm, I kind of want to agree. But I can see keypoints as still being relevant even if there's a superior tag. If a user searches for skeleton, they'll hopefully see all templates related to that. If we deprecate keypoints with the release of vector, it does make sense to remove keywords that may point to this... or simply remove this template altogether.

caitlinwheeless replied:
Fair enough!

caitlinwheeless commented on label_studio/annotation_templates/computer-vision/medical-image-classification-with-bboxes/config.yml:
```diff
@@ -2,7 +2,14 @@ title: Medical Image Classification with Bounding Boxes
+  <h1>Classify medical images and draw regions</h1>
+  <dl>
+    <dt>Industry Applications</dt>
+    <dd>radiology, pathology, oncology, diagnostic imaging, medical diagnosis, clinical decision support, medical AI, healthcare AI, X-ray, CT scan, MRI, ultrasound, mammography, histopathology, digital pathology, medical imaging, DICOM, radiological imaging, cancer detection, tumor segmentation, lesion detection, abnormality detection, screening programs, early detection, medical screening</dd>
```
Should we list DICOM if we don't actually support it?

caitlinwheeless commented on label_studio/annotation_templates/conversational-ai/intent-classification-and-slot-filling/config.yml:
```diff
@@ -2,7 +2,16 @@ title: Intent Classification and Slot Filling
 image: /static/templates/intent-classification-and-slot-filling.png
-details: <h1>Build task-oriented dialogue system by selecting dialogue intents and extracting slot entities</h1>
+details: |
+  <h1>Build task-oriented dialogue system by selecting dialogue intents and extracting slot entities</h1>
```
```suggestion
  <h1>Build a task-oriented dialogue system by selecting dialogue intents and extracting slot entities</h1>
```

caitlinwheeless commented on label_studio/annotation_templates/generative-ai/chatbot-assessment/config.yml:
```diff
@@ -3,7 +3,16 @@ type: community
+    <dt>Associated Models</dt>
+    <dd>human feedback collection, constitutional AI, safety evaluation, bias detection</dd>
+    <dt>Domain Terminology</dt>
+    <dd>gen ai, response safety, AI hallucination detection, conversational quality</dd>
```
```suggestion
    <dd>gen AI, response safety, AI hallucination detection, conversational quality</dd>
```

caitlinwheeless commented on label_studio/annotation_templates/generative-ai/chatbot-assessment/config.yml:
```diff
@@ -3,7 +3,16 @@ type: community
 image: /static/templates/generative-chatbot-assessment.png
-details: Collecting expert assessments of LLM-generated responses for quality assurance and model evaluation. This stage involves evaluating the model's responses based on criteria like correctness, relevance, and safety, which provides valuable feedback for further model refinement and helps ensure the final LLM meets desired performance standards.
+details: |
+  Collecting expert assessments of LLM-generated responses for quality assurance and model evaluation. This stage involves evaluating the model's responses based on criteria like correctness, relevance, and safety, which provides valuable feedback for further model refinement and helps ensure the final LLM meets desired performance standards.
```
No `<H1>` for these ones?

ricardoantoniocm replied:
Seems like they never had them, @caitlinwheeless . I'll add them.

ricardoantoniocm replied:
Added the headers to the templates that didn't have em.

caitlinwheeless commented on label_studio/annotation_templates/generative-ai/human-feedback-collection/config.yml:
```diff
@@ -3,7 +3,16 @@ type: community
+    <dt>Associated Models</dt>
+    <dd>RLHF, reward modeling, preference learning, pairwise comparison, constitutional AI</dd>
+    <dt>Domain Terminology</dt>
+    <dd>gen ai, human feedback, preference elicitation, reward signal, alignment research, value learning</dd>
```
```suggestion
    <dd>gen AI, human feedback, preference elicitation, reward signal, alignment research, value learning</dd>
```

caitlinwheeless commented on label_studio/annotation_templates/generative-ai/llm-ranker/config.yml:
```diff
@@ -3,7 +3,16 @@ type: community
 image: /static/templates/llm-ranker.png
-details: Here is a demo that showcases how to construct an interface enabling users to drag, arrange, and rank responses from different Large Language Models (LLMs) and chatbot systems based on their relevance to a specified prompt and their perceived bias, thus facilitating an intuitive and visual understanding of how well these AI systems align with the topic and the degree of subjective interp ...
+details: |
+  Here is a demo that showcases how to construct an interface enabling users to drag, arrange, and rank responses from different Large Language Models (LLMs) and chatbot systems based on their relevance to a specified prompt and their perceived bias, thus facilitating an intuitive and visual understanding of how well these AI systems align with the topic and the degree of subjective interpretatio ...
```
Did you change how you want to do this text? Previous ones were shorter and wrapped in h1 tags

caitlinwheeless commented on label_studio/annotation_templates/generative-ai/llm-ranker/config.yml:
```diff
@@ -3,7 +3,16 @@ type: community
+    <dt>Associated Models</dt>
+    <dd>RLHF, reward modeling, constitutional AI, alignment, fine-tuning, instruction tuning</dd>
+    <dt>Domain Terminology</dt>
+    <dd>gen ai, human feedback, preference learning, prompt engineering</dd>
```
```suggestion
    <dd>gen AI, human feedback, preference learning, prompt engineering</dd>
```

caitlinwheeless commented on label_studio/annotation_templates/generative-ai/response-grading/config.yml:
```diff
@@ -3,7 +3,16 @@ type: community
+    <dt>Associated Models</dt>
+    <dd>human evaluation, quality metrics, response scoring, rubric-based assessment</dd>
+    <dt>Domain Terminology</dt>
+    <dd>gen ai, response quality, content grading, automated assessment, performance rating</dd>
```
```suggestion
    <dd>gen AI, response quality, content grading, automated assessment, performance rating</dd>
```

caitlinwheeless commented on label_studio/annotation_templates/generative-ai/supervised-llm/config.yml:
```diff
@@ -3,7 +3,16 @@ type: community
+    <dt>Associated Models</dt>
+    <dd>supervised fine-tuning, instruction following, prompt-response pairs, few-shot learning</dd>
+    <dt>Domain Terminology</dt>
+    <dd>gen ai, SFT, instruction tuning, model specialization, task adaptation, performance optimization</dd>
```
```suggestion
    <dd>gen AI, SFT, instruction tuning, model specialization, task adaptation, performance optimization</dd>
```

caitlinwheeless commented on label_studio/annotation_templates/generative-ai/visual-ranker/config.yml:
```diff
@@ -3,7 +3,16 @@ type: community
+    <dt>Associated Models</dt>
+    <dd>DALL-E, Midjourney, Stable Diffusion, Firefly, generative adversarial networks</dd>
+    <dt>Domain Terminology</dt>
+    <dd>gen ai, visual quality ranking, image generation comparison, creative AI evaluation, aesthetic assessment, visual preference modeling</dd>
```
```suggestion
    <dd>gen AI, visual quality ranking, image generation comparison, creative AI evaluation, aesthetic assessment, visual preference modeling</dd>
```

caitlinwheeless commented on label_studio/annotation_templates/natural-language-processing/taxonomy/config.yml:
```diff
@@ -3,7 +3,16 @@ type: community
 image: /static/templates/taxonomy.png
-details: <h1>Do multilabel hierarchical classification</h1>
+details: |
+  <h1>Do multilabel hierarchical classification</h1>
```
```suggestion
  <h1>Multilabel hierarchical classification</h1>
```

caitlinwheeless commented on label_studio/annotation_templates/time-series-analysis/time-series-forecasting/config.yml:
```diff
@@ -2,7 +2,14 @@ title: Time Series Forecasting
+    <dd>stock prediction, financial markets, trading algorithms, price prediction, cryptocurrency, forex, market trends, algorithmic trading, IoT sensors, sensor data, predictive maintenance, equipment monitoring, energy consumption, smart grid, temperature monitoring, demand forecasting, sales prediction, inventory management, supply chain, resource planning, capacity planning, budget forecastin ...
+    <dt>Associated Models</dt>
+    <dd>ARIMA, LSTM, Prophet, seasonal decomposition, trend analysis</dd>
+  </dl>
```
No domain terminology for this one?

ricardoantoniocm committed:
Update label_studio/annotation_templates/conversational-ai/intent-classification-and-slot-filling/config.yml

Co-authored-by: C L W <caitlin@humansignal.com>

ricardoantoniocm committed:
Remove DICOM from Medical Image Classification with Bounding Boxes. This model is not related to simple image bounding boxes and is also not supported.

ricardoantoniocm committed:
Merge branch 'fb-fit-652' of https://github.com/heartexlabs/label-studio into fb-fit-652

ricardoantoniocm committed:
Update label_studio/annotation_templates/generative-ai/supervised-llm/config.yml

Co-authored-by: C L W <caitlin@humansignal.com>

ricardoantoniocm committed:
Update label_studio/annotation_templates/generative-ai/response-grading/config.yml

Co-authored-by: C L W <caitlin@humansignal.com>

ricardoantoniocm committed:
Update label_studio/annotation_templates/generative-ai/llm-ranker/config.yml

Co-authored-by: C L W <caitlin@humansignal.com>

ricardoantoniocm committed:
Adds missing headers. Change gen ai to gen AI.

ricardoantoniocm committed:
Merge branch 'fb-fit-652' of https://github.com/heartexlabs/label-studio into fb-fit-652

ricardoantoniocm committed:
Adds domain terminology to Time Series Forecasting template.

ricardoantoniocm committed:
Update label_studio/annotation_templates/natural-language-processing/taxonomy/config.yml

Co-authored-by: C L W <caitlin@humansignal.com>

ricardoantoniocm committed:
Shortens ranker-related headers.

ricardoantoniocm committed:
Update label_studio/annotation_templates/computer-vision/keypoints/config.yml

Co-authored-by: C L W <caitlin@humansignal.com>

ricardoantoniocm committed:
Remove digital asset management from keywords for NLP template.

ricardoantoniocm committed:
Remove the word Provide from text summarization template.

ricardoantoniocm committed:
Simplify headers in details, avoiding the word "Provide".

yyassi-heartex reviewed (approved):
updating yml files to improve the search results. looks good to me!

ricardoantoniocm committed:
Adds NLP keyword to associated templates.

ricardoantoniocm committed:
Adds keywords to Time Series and Tabular Data templates.

ricardoantoniocm commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/17600153036)
> Successfully merged:  create mode 100644 web/libs/editor/src/tags/object/Chat.js

ricardoantoniocm commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/17600160800)

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-fit-652'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17600153036

ricardoantoniocm committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17600160800
